misago/legal/models.py

```diff
@@ -1 +1,127 @@
 # Empty models file that triggers migrations for this app
+from django.db import models
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.utils.translation import ugettext_lazy as _
+
+import datetime
+
+
+class NoActiveLegalAgreement(ValidationError):
+    pass
+
+
+class BaseModel(models.Model):
+    created = models.DateTimeField(auto_now_add=True, editable=False)
+    modified = models.DateTimeField(auto_now=True, editable=False)
+
+    class Meta:
+        abstract = True
+
+
+class LegalAgreementManager(models.Manager):
+    def get_current(self):
+        try:
+            return self.get(active=True)
+        except self.model.DoesNotExist:
+            raise NoActiveLegalAgreement(
+                u'Please create an active LegalAgreement'
+            )
+
+
+class LegalAgreement(BaseModel):
+    IMP_HIGH = 2
+    IMP_MEDIUM = 1
+    IMP_LOW = 0
+    IMPORTANCE = (
+        (IMP_HIGH, 'High'),
+        (IMP_MEDIUM, 'Medium'),
+        (IMP_LOW, 'Low'),
+    )
+    TOS = 'terms_of_service'
+    PRIVACY = 'privacy_policy'
+    Legal = (
+        (TOS, 'Terms of service'),
+        (PRIVACY, 'Privacy Policy'),
+    )
+    active = models.BooleanField(
+        default=False,
+        verbose_name=_('active'),
+        help_text=_(
+            u'Only one terms of service is allowed to be active'
+        )
+    )
+    title = models.CharField(
+        max_length=20,
+        choices=Legal,
+        default=TOS
+    )
+    published = models.DateTimeField(editable=False, null=True)
+    importance = models.IntegerField(
+        choices=IMPORTANCE,
+        default=IMP_HIGH,
+    )
+    text = models.TextField(verbose_name=_('content'), blank=True)
+    objects = LegalAgreementManager()
+
+    def latest_published(self):
+        return LegalAgreementVersion.objects.filter(title=self.title).order_by('-version')
+
+    def save(self, *args, **kwargs):
+        """ Ensure that only LegalAgreement is active"""
+        if self.active:
+            LegalAgreement.objects.exclude(id=self.id).update(active=False)
+        else:
+            if not LegalAgreement.objects \
```

Comment:
Misago doesn't use `\` to break long lines in our codebase. Also as general rule, please avoid longly chained expressions within `if` conditions. Instead assign such querysets to variables like `active_agreements = LegalAgreement.objects...`.